client/src/controllers/AutoSizeController.ts

```diff
@@ -0,0 +1,40 @@
+import { Controller } from '@hotwired/stimulus';
+import autosize from 'autosize';
+import { debounce } from '../utils/debounce';
+
+/**
+ * Adds the ability for a text area element to be auto-sized as the user
+ * types in the field so that it expands to show all content.
+ *
+ * @example
+ * <textarea data-controller="w-autosize"></textarea>
+ */
+
+export default class AutosizeController extends Controller<HTMLTextAreaElement> {
+  resizeObserver?: ResizeObserver;
+
+  resize() {
+    autosize.update(this.textareaTarget);
+  }
+
+  initialize() {
+    this.resize = debounce(this.resize.bind(this), 50);
+  }
+
+  connect() {
+    autosize(this.textareaTarget);
+    this.resizeObserver = new ResizeObserver(this.resize);
+    this.resizeObserver.observe(this.textareaTarget);
+  }
+
+  disconnect() {
+    this.resizeObserver?.disconnect();
+    autosize.destroy(this.textareaTarget);
+  }
+
+  get textareaTarget(): HTMLTextAreaElement {
```

Comment:
This is the incorrect way to use a Stimulus target, however this controller approach does not need targets as you can just use `this.element` to get the controlled element.

Reply:
For future reference. If you do need a target, you must declare the target name with `static targets = ['something'];`. 

Then you can use it in the controller via `this.somethingTarget`.

See https://stimulus.hotwired.dev/reference/targets